comp/core/autodiscovery/autodiscoveryimpl/autoconfig.go

```diff
@@ -94,6 +96,15 @@ type AutoConfig struct {
 	ranOnce *atomic.Bool
 }
 
+const (
+	// MaxWmetaWaitTime is the maximum time to wait for wmeta being ready
+	MaxWmetaWaitTime = 10 * time.Second
+	// WmetaCheckMinInterval is the initial interval to check if wmeta is ready
+	WmetaCheckMinInterval = 50 * time.Microsecond
+	// WmetaCheckMaxInterval is the maximum interval to check if wmeta is ready
+	WmetaCheckMaxInterval = 1 * time.Second
+)
```

Comment:
Just a bit unsure regarding those values as I feel like the initial interval is too small. Another (very small) nit would be to re-order/rename constants for added clarity.
```suggestion
const (
	// WmetaCheckInitialInterval is the initial interval to check if wmeta is ready
	WmetaCheckInitialInterval = 20 * time.Millisecond
	// WmetaCheckMaxInterval is the maximum interval to check if wmeta is ready
	WmetaCheckMaxInterval = 1 * time.Second
	// WmetaMaxElapsedTime is the maximum time to wait for wmeta being ready
	WmetaMaxElapsedTime = 10 * time.Second
)
```

Reply:
Agree with @wdhif 
Here we want to add a retrier on actions that requires external calls, so interval with an order of a few millisecond duration make more sense than microsecond.